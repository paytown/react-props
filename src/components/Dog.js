import React from 'react';
import PropTypes from 'prop-types';

function Dog({ name, age, sign }) {
  return (
    <main>
      <h3>Name</h3>
      <p>{name}</p>

      <h3>Age</h3>
      <p>{age ? age : 'unknown'}</p>

      <h3>Sign</h3>
      <p>{sign}</p>
    </main>
  );
}

Dog.propTypes = {
  name: PropTypes.string.isRequired,
  age: PropTypes.number,
  sign: PropTypes.string.isRequired,
};

export default Dog;
