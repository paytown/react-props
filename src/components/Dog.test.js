import React from 'react';
import { shallow } from 'enzyme';
import Dog from './Dog';

describe('Doggie', () => {
  it('let\'s see that dog', () => {
    const wrapper = shallow(<Dog />);
    expect(wrapper).toMatchSnapshot();
  });
});
