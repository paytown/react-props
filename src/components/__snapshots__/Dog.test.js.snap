// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Doggie let's see that dog 1`] = `
<main>
  <h3>
    Name
  </h3>
  <p />
  <h3>
    Age
  </h3>
  <p>
    unknown
  </p>
  <h3>
    Sign
  </h3>
  <p />
</main>
`;
